test/rekt/features/kafka_source.go

```diff
@@ -248,3 +280,376 @@ func compareConsumerGroup(source string, cmp func(*internalscg.ConsumerGroup) er
+	)
+}
+
+func marshalJSON(val interface{}) string {
+	data, _ := json.Marshal(val)
+	return string(data)
+}
+
+func KafkaSourceWithEventAfterUpdate(kafkaSource string, kafkaSink string) *feature.Feature {
+
+	f := feature.NewFeatureNamed("KafkaSourceWithEventAfterUpdate")
+
+	receiver := feature.MakeRandomK8sName("eventshub-receiver")
+	sender := feature.MakeRandomK8sName("eventshub-sender")
+
+	f.Setup("install eventshub receiver", eventshub.Install(receiver, eventshub.StartReceiver))
+
+	update := kafkaSourceUpdate{
+		sink:     receiver,
+		ordering: sourcesv1beta1.Unordered,
+		auth: authSetup{
+			bootstrapServers: testingpkg.BootstrapServersPlaintextArr,
+			TLSEnabled:       false,
+			SASLEnabled:      false,
+		},
+	}
+	f.Setup("update kafka source", ModifyKafkaSource(kafkaSource, update))
+	f.Setup("kafka source is ready", kafkasource.IsReady(kafkaSource))
+
+	e := cetest.FullEvent()
+
+	options := []eventshub.EventsHubOption{
+		eventshub.StartSenderToResource(kafkasink.GVR(), kafkaSink),
+		eventshub.AddSequence,
+		eventshub.SendMultipleEvents(1, time.Millisecond),
+		eventshub.InputEvent(e),
+	}
+
+	f.Requirement("install eventshub sender for kafka sink", eventshub.Install(sender, options...))
+
+	matcher := AllOf(
+		HasData(e.Data()),
+		HasTime(e.Time()),
+	)
+	f.Assert("sink receives event", matchEvent(receiver, matcher))
+
+	return f
+}
+
+func ModifyKafkaSource(kafkaSourceName string, update kafkaSourceUpdate) feature.StepFn {
+	return func(ctx context.Context, t feature.T) {
+		ksObj, err := waitForKafkaSourceToBeReconciled(ctx, kafkaSourceName)
+		if err != nil {
+			t.Fatalf("Failed waiting for KafkaSource to be reconciled: %v", ksObj)
+		}
+
+		// Update KafkaSource spec
+		ksObj.Spec.Sink.Ref = service.AsKReference(update.sink)
+		ksObj.Spec.KafkaAuthSpec.BootstrapServers = update.auth.bootstrapServers
+		ksObj.Spec.KafkaAuthSpec.Net.TLS.Enable = update.auth.TLSEnabled
+		ksObj.Spec.KafkaAuthSpec.Net.SASL.Enable = update.auth.SASLEnabled
+		ksObj.Spec.Ordering = &update.ordering
+
+		err = updateKafkaSource(ctx, ksObj)
+		if err != nil {
+			t.Fatalf("Failed to update v1beta1 KafkaSource %q: %v", ksObj.Name, err)
+		}
+
+		_, err = waitForKafkaSourceToBeReconciled(ctx, kafkaSourceName)
+		if err != nil {
+			t.Fatalf("Failed waiting for KafkaSource to be reconciled: %v", ksObj)
+		}
+	}
+}
+
+func waitForKafkaSourceToBeReconciled(ctx context.Context, kafkaSourceName string) (*sourcesv1beta1.KafkaSource, error) {
+	var ksObj *sourcesv1beta1.KafkaSource
+	interval, timeout := environment.PollTimingsFromContext(ctx)
+	err := wait.Poll(interval, timeout, func() (done bool, err error) {
+		ns := environment.FromContext(ctx).Namespace()
+		ksObj, err = kafkaclient.Get(ctx).SourcesV1beta1().
+			KafkaSources(ns).
+			Get(ctx, kafkaSourceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return ksObj.Status.IsReady() && ksObj.Status.ObservedGeneration == ksObj.Generation, nil
+	})
+	return ksObj, err
+}
```

Comment:
isn't this the same as kafkasource.IsReady?

Reply:
You're right